components/connector/internal/readiness/handler.go

```diff
@@ -0,0 +1,34 @@
+package readiness
+
+import (
+	"net/http"
+
+	"k8s.io/client-go/discovery"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+func NewHTTPHandler(log *logrus.Logger, apiServerClient discovery.ServerVersionInterface) func(writer http.ResponseWriter, request *http.Request) {
+	writeResponseFunc := handleResponseFunc(log)
+	return func(writer http.ResponseWriter, request *http.Request) {
+		_, err := apiServerClient.ServerVersion()
+		if err != nil {
+			logrus.Errorf("Failed to access API Server: %s.", err.Error())
```

Comment:
Since `log` is provided to `NewHTTPHandler`, maybe reuse `log` here (and below) instead of `logrus`, to keep things consistent throughout the function.